bundler/lib/bundler/retry.rb

```diff
@@ -51,6 +54,18 @@ def fail_attempt(e)
       return true unless name
```

Comment:
I find it a bit strange that we run the backoff strategy only when there is a `name`.

I understand that the early return is only reached in tests to prevent outputting log, but now that we are adding a new behavior, I'd expect it to run all the time regardless if there is name.